File: app/src/main/java/io/github/ovso/righttoknow/data/network/VideoService.java
package io.github.ovso.righttoknow.data.network;

import io.github.ovso.righttoknow.data.network.model.video.Search;
import io.reactivex.Single;
import java.util.Map;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface VideoService {
  @GET("youtube/v3/search")
  Single<Search> getResult(@QueryMap Map<String, Object> queryMap);
}

File: app/src/main/java/io/github/ovso/righttoknow/data/network/VideoService.kt
package io.github.ovso.righttoknow.data.network

import io.github.ovso.righttoknow.data.network.model.video.Search
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface VideoService {
  @GET("youtube/v3/search")
  fun getResult(@QueryMap queryMap: Map<String, Any>): Single<Search>
}
